src/core/library.c

```diff
@@ -889,6 +894,75 @@ QuicLibrarySetGlobalParam(
 
         break;
 
+    case QUIC_PARAM_GLOBAL_RAW_DATAPATH_PROCS: {
+        if (Buffer == NULL) {
+            if (MsQuicLib.RawDataPathProcList != NULL) {
+                CXPLAT_FREE(MsQuicLib.RawDataPathProcList, QUIC_POOL_RAW_DATAPATH_PROCS);
+                MsQuicLib.RawDataPathProcList = NULL;
+                MsQuicLib.RawDataPathProcListLength = 0;
+            }
+            Status = QUIC_STATUS_SUCCESS;
+            break;
+        }
+
+        if (BufferLength < sizeof(uint16_t) || BufferLength % sizeof(uint16_t) != 0) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (MsQuicLib.Datapath != NULL) {
+            QuicTraceLogError(
+                LibraryRawDataPathProcsSetAfterDataPathInit,
+                "[ lib] Tried to change raw datapath procs after datapath initialization");
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        uint32_t RawDataPathProcListLength = BufferLength / sizeof(uint16_t);
+        uint16_t* Cpus = (uint16_t*)Buffer;
+        _Analysis_assume_(BufferLength >= 4); // SAL incorrectly thinks 4 bytes will be accessed.
```

Comment:
Whaaaat? How?!

Reply:
```
D:\msquic\src\core\library.c(1016,1): error C2220: the following warning is treated as an error [D:\msquic\build\windows\x64_schannel\src\core\core.vcxproj]
D:\msquic\src\core\library.c(925): warning C6385: Reading invalid data from 'Cpus':  the readable size is 'BufferLength' bytes, but '4' bytes may be read.: Lines:
 921, 736, 737, 739, 897, 898, 908, 913, 921, 922, 923, 924, 925, 924, 925 [D:\msquic\build\windows\x64_schannel\src\core\core.vcxproj]
```

I don't really know how. It's either SAL is buggy or this code is buggy but the code looks correct to me.

Reply:
Ok. That doesn't make sense. I don't see anything wrong with the code.

Reply:
Yeah, this is absolutely something wrong with SAL. Replacing the indexer with a direct pointer access also avoids the warning.

`if (*(Cpus + i) >= CxPlatProcActiveCount()) {` 

The above works... (And is the exact same code...)

Reply:
Nice, I will work around it by doing that.